cmd/query.go

```diff
@@ -1105,24 +1151,27 @@ $ %s query clients-expiration demo-path`,
 				return errDst
 			}
 
-			// if only the src light client is found, just print info for source light client
-			if errSrc == nil && errDst != nil {
-				fmt.Fprintln(cmd.OutOrStdout(), relayer.SPrintClientExpiration(c[src], srcExpiration, srcClientInfo))
-				return nil
-			}
+			output, _ := cmd.Flags().GetString(flagOutput)
 
-			// if only the dst light client is found, just print info for destination light client
-			if errDst == nil && errSrc != nil {
-				fmt.Fprintln(cmd.OutOrStdout(), relayer.SPrintClientExpiration(c[dst], dstExpiration, dstClientInfo))
-				return nil
-			}
+			srcClientExpiration := relayer.SPrintClientExpiration(c[src], srcExpiration, srcClientInfo)
+			dstClientExpiration := relayer.SPrintClientExpiration(c[dst], dstExpiration, dstClientInfo)
 
-			fmt.Fprintln(cmd.OutOrStdout(), relayer.SPrintClientExpiration(c[src], srcExpiration, srcClientInfo))
-			fmt.Fprintln(cmd.OutOrStdout(), relayer.SPrintClientExpiration(c[dst], dstExpiration, dstClientInfo))
+			if output == "json" {
+				srcClientExpiration = relayer.SPrintClientExpirationJson(c[src], srcExpiration, srcClientInfo)
+				dstClientExpiration = relayer.SPrintClientExpirationJson(c[dst], dstExpiration, dstClientInfo)
+			}
 
+			if errSrc == nil && errDst == nil {
+				fmt.Fprintln(cmd.OutOrStdout(), srcClientExpiration)
+				fmt.Fprintln(cmd.OutOrStdout(), dstClientExpiration)
+			} else if errSrc == nil && errDst != nil {
+				fmt.Fprintln(cmd.OutOrStdout(), dstClientExpiration)
+			} else if errDst == nil && errSrc != nil {
+				fmt.Fprintln(cmd.OutOrStdout(), srcClientExpiration)
+			}
```

Comment:
we could eliminate one branch on this by only checking for the cases where we print one client's information e.g.

```go
if errSrc == nil && errDst != nil {
    fmt.Fprintln(cmd.OutOrStdout(), dstClientExpiration)
} else if errDst == nil && errSrc != nil {
    fmt.Fprintln(cmd.OutOrStdout(), srcClientExpiration)
}

fmt.Fprintln(cmd.OutOrStdout(), srcClientExpiration)
fmt.Fprintln(cmd.OutOrStdout(), dstClientExpiration)
```

this is just a suggestion but it feels slightly more readable imo

Reply:
I just realized while reading this . 
would this  be optimal ?

if errSrc == nil {
    fmt.Fprintln(cmd.OutOrStdout(), srcClientExpiration)
}

 if errDst == nil {
    fmt.Fprintln(cmd.OutOrStdout(), dstClientExpiration)
}

Reply:
oh yeah this is even better, good call!